### DISCUSSION TITLE
how to hide `Ambiguous Alternatives Detected` related message

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1560] [Q&A]
if i didn't want to print this message,any method to hide it?i found related code at `chevrotain`,but didn't konw how langium refer it
![image](https://github.com/eclipse-langium/langium/assets/26995040/10282473-bf81-455d-8fda-c462bd1caeef)

**Comment by msujew** [comment]
Hey @zrwsmd,

I would recommend you to fix the root issue before attempting to hide the warning. Parser ambiguities lead to performance issues down the line.

**Reply by msujew** [reply]
@zrwsmd Can you share the `PrimaryExpression` rule (and associated rules) so we can help you figure out how to resolve the ambiguity?

**Reply by zrwsmd** [reply]
> @zrwsmd Can you share the `PrimaryExpression` rule (and associated rules) so we can help you figure out how to resolve the ambiguity?

[demo.txt](https://github.com/user-attachments/files/15969111/demo.txt)

**Reply by zrwsmd** [reply]
`Constant`  rule refer to https://github.com/Serhioromano/vscode-st/blob/b3f9b17b141e6186c32d938fd00f85ada046df7a/src/language-server/st.langium

**Reply by msujew** [reply]
Ok, can you also share some text to reproduce the ambiguity? The grammar is fairly convoluted and it's hard to figure out how to reproduce this.
